cmd/dt/unwrap/unwrap.go

```diff
@@ -508,6 +527,7 @@ func NewCmd(cfg *config.Config) *cobra.Command {
 
 	cmd.PersistentFlags().StringVar(&version, "version", version, "when unwrapping remote Helm charts from OCI, version to request")
 	cmd.PersistentFlags().StringVar(&pushChartURL, "push-chart-url", pushChartURL, "push the unwrapped Helm chart to the given URL")
+	cmd.PersistentFlags().StringVar(&pushRepository, "push-repository", pushRepository, "the repository to be used instead for write operations e.g (demo.goharbor.io/write/test_repo)")
```

Comment:
I guess a better parameter name would be `push-container-registry`, right? I mean, in a given chart there are N potential repositories published in a single registry/namespace, see:

- https://learn.microsoft.com/en-us/azure/container-registry/container-registry-concepts

Reply:
Maybe. Unclear to me as the current flag for helm chart is `push-chart-url` ....

Reply:
https://github.com/vmware-labs/distribution-tooling-for-helm/issues/93#issuecomment-2385070657

Reply:
as oci repos are general purpose not specific to store binary images, charts, or just a file just registry might be a good option

Reply:
I'd go for sth like this:
```golang
	cmd.PersistentFlags().StringVar(&pushContainerRegistry, "push-container-registry", pushContainerRegistry, "push the container images included in the unwrapped Helm chart to the given container registry (including namespace)")
```
What do you think @mpermar?

Reply:
I think registry is more appropriate than repository indeed. Similarly registries have long move from containers alone. I think a good name is `push-registry`.

Reply:
Agree. I will attempt have this clean up tomorrow. Thanks

Reply:
Actually I take it back @machecazzon @juan131. The value can be both a registry and a repository so probably would need some other name like `push-uri`. 

But, actually, while looking in detail to the naming, and this is totally my bad as I am lost touch with this project and it has evolved quite a while since I left it, I can see there is a `push-chart-url` which I believe it might do exactly what this PR is trying to do.

Reply:
Have been thinking about this a bit more and I believe that having `--values-registry` could be more appropriate as is actually what I am looking for. 

> "Have the values of the chart to be relocated not to the unwrap target but instead the value provided by e.g `--values-registry`." 

That's more like what I did on my first attempt to cover the use case.

Right now I have a custom build with what I need to have the desired outputs that includes this and some other changes and is not a big deal but I would prefer that this tool could support this behaviour.

Back into the naming, kaniko has this switch and it's format somehow make me think of this.

https://github.com/GoogleContainerTools/kaniko?tab=readme-ov-file#flag---registry-map

Very likely I will abandon this PR and come back later with a better solution and better PR hopefully, but if wanted can still try to address this PR comments.

In regards the `push-chart-url` I am currently using that switch to tune the chart destination path as well as needing this same "feature". This is related to some other "issue" I have with current implementation and patched for good on my end :)

> RelocateImageURL method that seems that it  might change the original path and have something else on the unwrap target depending on the input provided and image url is. This turns that image path not be the exactly the same on the destination as the original source what I need to ensure preservation.

My current scenario: GAR Reader / GAR Writer

* GAR Reader (Virtual REpo) is a able to pull from Gitlab registry
* dev teams build their images and push to Gitlab registry
* dt is used to build a wrap on ci  ensuring images exist, they are pinned (lock/verify). Skip save images that source from Gitlab but still map then on Images.lock
* the unwrap is done to gitlab what relocates chart values to be GAR reader repo and not Gitlab and basically skips the images push but ensure they exist and digest matches...
*  Once a "golden" wrap chart is built from gitlab, it's unwrapped to GAR Write repo (that is configured to have immutable tags) chart  + all docker images references using dt. When chart is published this "golden wrap" takes over and is what's will be consumed via argocd if that version is requested cutting the link with Gitlab. 
* The chart values are again preserved to be the GAR Reader repo and not the  GAR Writer that is the unwrap target

For all the previous to work transparently It's crucial to to guarantee images path preservation from Gitlab to GAR is ensured.

Reply:
That's great context. And wow, you have built a thing there.

I'm approaching your PR here from a point of view of trying to keep the tool as simplest as possible. Ideally I'd like to have very simple building blocks even when that might come at the expense of having extra complexity in the client scripts. It makes this project more sustainable in the long run. 

From that perspective I see that `helm dt unwrap target-images-registry --push-chart-url=target-chart-registry` is giving us two basic primitive arguments: one to set where images are going to be pushed, and a second one to set where the chart is going to be pushed. It can certainly be improved (e.g. kaniko for multi-registry images), but it feels effective enough.

But at the same time I have to be honest here, I'm humbled by your use case and what you are doing. So, I'm not sure if I'm missing anything obvious. Perhaps the right thing to do is as you say to abandon this one and then try to  approach it with some simpler baby-step kind of change. Not sure, tbh.

Reply:
Hello @mpermar 

Sorry for delay getting back.

I ended with something like this

* build a wrap (from local filesystem "helm chart") that does not pull images but just contain helm chart and Images.lock. Unwrap will basically just push the chart  (source registry and target registry are the same) gitlab --> gitlab

* with the previous wrap (no local images) be able to unwrap to destination registry but copy the referenced docker images as usual while source images instead of being local filesystem be a crane copy from registry source to target (source registry and target registry are not the same) .. in my case gitlab --> gar